uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java

```diff
@@ -52,54 +49,70 @@ public class LocalDeploymentSteps {
-            final AWSResources resources,
             final ComponentOverrides overrides,
             final TestContext testContext,
             final ComponentPreparationService componentPreparation,
             final ScenarioContext scenarioContext,
             final WaitSteps waits,
             final ObjectMapper mapper,
-            final Platform platform,
             final Device device) {
-        this.resources = resources;
         this.overrides = overrides;
         this.testContext = testContext;
         this.componentPreparation = componentPreparation;
         this.scenarioContext = scenarioContext;
         this.waits = waits;
         this.mapper = mapper;
-        this.platform = platform;
         this.artifactPath = testContext.installRoot().resolve(LOCAL_STORE).resolve(ARTIFACTS_DIR);
         this.recipePath = testContext.installRoot().resolve(LOCAL_STORE).resolve(RECIPE_DIR);
         this.device = device;
     }
 
+    /**
+     * implemented the step of installing a custom component with configuration.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
     @When("I install the component {word} from local store with configuration")
     public void installComponentWithConfiguration(final String componentName, final DataTable configurationTable)
             throws InterruptedException, IOException {
         List<Map<String, String>> configuration = configurationTable.asMaps(String.class, String.class);
-        List<String> componentSpecs = new ArrayList<>(Arrays.asList(
+        List<String> componentSpecs = Arrays.asList(
                 componentName, LOCAL_STORE_RECIPES.resolve(String.format("%s.yaml", componentName)).toString()
-        ));
+        );
         installComponent(componentSpecs, configuration);
     }
 
+    /**
+     *  implemented the step of updating a custom component's configuration.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I update the component {word} with configuration")
+    public void updateComponentWithConfiguration(final String componentName, final DataTable configurationTable)
+            throws InterruptedException, IOException {
+        List<Map<String, String>> configuration = configurationTable.asMaps(String.class, String.class);
+        updateComponent(componentName, configuration);
```

Comment:
Rename this to `updateComponentConfiguration`

Reply:
done